File: investwise_android/lib/domain/entities/global_stats.dart
class GlobalStats {
  final double totalDeposits;
  final double investedCapital;
  final int totalMembers;
  final int totalShares;
  final double yieldIndex;
  final double fundStability;
  final List<TrendDataPoint> trendData;
  final TopInvestor topInvestor;

  GlobalStats({
    required this.totalDeposits,
    required this.investedCapital,
    required this.totalMembers,
    required this.totalShares,
    required this.yieldIndex,
    required this.fundStability,
    required this.trendData,
    required this.topInvestor,
  });
}

class TrendDataPoint {
  final String month;
  final double inflow;
  final double outflow;

  TrendDataPoint({
    required this.month,
    required this.inflow,
    required this.outflow,
  });
}

class TopInvestor {
  final String name;
  final String role;

  TopInvestor({
    required this.name,
    required this.role,
  });
}

File: server/models/GlobalStats.js
import mongoose from 'mongoose';

const GlobalStatsSchema = new mongoose.Schema({
    totalDeposits: { type: Number, default: 0 },
    investedCapital: { type: Number, default: 0 },
    totalMembers: { type: Number, default: 0 },
    totalShares: { type: Number, default: 0 },
    yieldIndex: { type: Number, default: 0 },

    // Monthly trend data for the chart (last 12 months)
    trendData: [{
        month: String, // e.g. "Jan 2024"
        inflow: { type: Number, default: 0 },
        outflow: { type: Number, default: 0 }
    }],

    // Sector distribution for the pie chart
    sectorDiversification: [{
        category: String,
        value: { type: Number, default: 0 }
    }],

    fundStability: { type: Number, default: 100 },

    lastUpdated: { type: Date, default: Date.now }
}, { timestamps: true });

const GlobalStats = mongoose.model('GlobalStats', GlobalStatsSchema);
export default GlobalStats;
